File: Lista/Java/Q12.java
class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public double getPreco() {
        return preco;
    }

    public Produto somar(Produto outroProduto) {
        double novoPreco = this.preco + outroProduto.getPreco();
        return new Produto("Combo de " + this.nome + " e " + outroProduto.nome, novoPreco);
    }

    public void exibirInformacoes() {
        System.out.println("Produto: " + nome + ", Preço: R$ " + preco);
    }
}

public class Main {
    public static void main(String[] args) {
        Produto produto1 = new Produto("Produto A", 50.00);
        Produto produto2 = new Produto("Produto B", 30.00);

        produto1.exibirInformacoes();
        produto2.exibirInformacoes();

        Produto combo = produto1.somar(produto2);

        combo.exibirInformacoes();
    }
}

File: Lista/Go/Q12.go
package main

import "fmt"

type Produto struct {
    Nome  string
    Preco float64
}

func SomarProdutos(p1, p2 Produto) Produto {
    return Produto{
        Nome:  p1.Nome + " & " + p2.Nome,
        Preco: p1.Preco + p2.Preco,
    }
}

func main() {
    produto1 := Produto{Nome: "Produto A", Preco: 100.0}
    produto2 := Produto{Nome: "Produto B", Preco: 150.0}

    produtoSoma := SomarProdutos(produto1, produto2)

    fmt.Printf("Produto Resultante: %s\nPreço Total: R$%.2f\n", produtoSoma.Nome, produtoSoma.Preco)
}
